pyccl/covariances.py

```diff
@@ -127,3 +129,240 @@ def angular_cl_cov_cNG(cosmo, cltracer1, cltracer2, ell, tkka, fsky=1.,
     return cov
+
+
+def sigma2_B_disc(cosmo, a=None, fsky=1., p_of_k_a=None):
+    """ Returns the variance of the projected linear density field
+    over a circular disc covering a sky fraction `fsky` as a function
+    of scale factor. This is given by
+
+    .. math::
+        \\sigma^2_B(z) = \\int_0^\\infty \frac{k\\,dk}{2\\pi}
+            P_L(k,z)\\,\\left[\\frac{2J_1(k R(z))}{k R(z)}\\right]^2,
+
+    where :math:`R(z)` is the corresponding radial aperture as a
+    function of redshift. This quantity is used to compute the
+    super-sample covariance.
+
+    Args:
+        cosmo (:class:`~pyccl.core.Cosmology`): a Cosmology object.
+        a_arr (float, array_like or `None`): an array of scale factor
+            values at which to evaluate the projected variance. If
+            `None`, a default sampling will be used.
+        fsky (float): sky fraction.
+        p_of_k_a (:class:`~pyccl.pk2d.Pk2D`, str, or `None`): Linear
+            power spectrum to use. Defaults to `None`.
+
+    Returns:
+        float or array_like: values of the projected variance.
+    """
+    status = 0
+    if a is None:
+        na = lib.get_pk_spline_na(cosmo.cosmo)
+        a_arr, status = lib.get_pk_spline_a(cosmo.cosmo, na, status)
+        check(status)
+    else:
+        a_arr = np.atleast_1d(a)
+        na = len(a_arr)
+
+    chi_arr = comoving_radial_distance(cosmo, a_arr)
+    R_arr = chi_arr * np.arccos(1-2*fsky)
+    psp = parse_pk2d(cosmo, p_of_k_a, is_linear=True)
+
+    s2B_arr, status = lib.sigma2b_vec(cosmo.cosmo, a_arr, R_arr, psp,
+                                      na, status)
+    check(status, cosmo=cosmo)
+    if a is None:
+        return a_arr, s2B_arr
+    else:
+        if np.ndim(a) == 0:
+            return s2B_arr[0]
+        else:
+            return s2B_arr
+
+
+def sigma2_B_from_mask(cosmo, a=None, mask_wl=None, p_of_k_a=None):
+    """ Returns the variance of the projected linear density field, given the
+        angular power spectrum of the footprint mask and scale factor.
+        This is given by
+
+    .. math::
+        \\sigma^2_B(z) = \\frac{1}{\\chi^2{z}}\\sum_\\ell
+            P_L(\\frac{\\ell+\\frac{1}{2}}{\\chi(z)},z)\\,
+            (2\\ell+1)\\sum_m W^A_{\\ell m} {W^B}^*_{\\ell m},
+
+    where :math:`W^A_{\\ell m}` and :math:`W^B_{\\ell m}` are the spherical
+    harmonics decomposition of the footprint masks of fields `A` and `B`,
+    normalized by their area.
+
+    Args:
+        cosmo (:class:`~pyccl.core.Cosmology`): a Cosmology object.
+        a (float, array_like or `None`): an array of scale factor
+            values at which to evaluate the projected variance.
+        mask_wl (array_like): Array with the angular power spectrum of the
+            masks. The power spectrum should be given at integer multipoles,
+            starting at :math:`\\ell=0`. The power spectrum is normalized
+            as :math:`(2\\ell+1)\\sum_m W^A_{\\ell m} {W^B}^*_{\\ell m}`.
```

Comment:
Maybe add that it's the user's responsibility to provide this up to a sufficiently high ell.

Reply:
Done.